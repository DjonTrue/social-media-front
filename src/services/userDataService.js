import axios from "axios";

export const userDataService = {
  getUserData: (username) => axios.get(`/users?username=${username}`).then(({ data }) => data),
};
