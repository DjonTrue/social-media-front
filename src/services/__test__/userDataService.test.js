import axios from "axios";
import { userDataService } from "../userDataService";

jest.mock("axios");

describe("API call is successful", () => {
  let response = null;

  beforeEach(() => {
    response = {
      data: {
        _id: "61d9d847c97cb8bdc4691674",
        username: "AlenaBfit",
        email: "AlenaBfit@gmail.com",
        profilePicture: "",
        coverPicture: "",
        followers: ["61de9ae792d7aa5da2105351"],
        followings: ["61de9ae792d7aa5da2105351"],
        isAdmin: false,
        createdAt: "2022-01-08T18:30:31.666Z",
        __v: 0,
      },
      status: "ok",
    };
  });

  afterEach(() => {
    response = null;
  });

  it("getDataService returned value", async () => {
    axios.get.mockResolvedValueOnce(response);
    const data = await userDataService.getUserData("61d9d847c97cb8bdc4691674");

    expect(axios.get).toBeCalledTimes(1);
    expect(axios.get).toHaveBeenCalledWith("/users?username=61d9d847c97cb8bdc4691674");
    expect(data).toEqual({
      _id: "61d9d847c97cb8bdc4691674",
      username: "AlenaBfit",
      email: "AlenaBfit@gmail.com",
      profilePicture: "",
      coverPicture: "",
      followers: ["61de9ae792d7aa5da2105351"],
      followings: ["61de9ae792d7aa5da2105351"],
      isAdmin: false,
      createdAt: "2022-01-08T18:30:31.666Z",
      __v: 0,
    });
    expect(data).toMatchSnapshot();
  });
});
