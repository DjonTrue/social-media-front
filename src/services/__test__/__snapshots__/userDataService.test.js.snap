// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API call is successful getDataService returned value 1`] = `
Object {
  "__v": 0,
  "_id": "61d9d847c97cb8bdc4691674",
  "coverPicture": "",
  "createdAt": "2022-01-08T18:30:31.666Z",
  "email": "AlenaBfit@gmail.com",
  "followers": Array [
    "61de9ae792d7aa5da2105351",
  ],
  "followings": Array [
    "61de9ae792d7aa5da2105351",
  ],
  "isAdmin": false,
  "profilePicture": "",
  "username": "AlenaBfit",
}
`;
